spec/rails_helper.rb

```diff
@@ -904,12 +904,14 @@ def using_session_with_localhost_resolution(name, &block)
 
   class TestCurrentUserProvider < Auth::DefaultCurrentUserProvider
     def log_on_user(user, session, cookies, opts = {})
-      session[:current_user_id] = user.id
+      # Try using the main session as `session` sometimes is a server session
+      (cookies.try(:request).try(:session) || session)[:current_user_id] = user.id
```

Comment:
Sometimes `cookies` doesn’t come from the request because we’re using some custom helpers (they’re quite old), but that way everything is still working as expected without having to change everything.